### PR TITLE
Adding AuthorizePage & AuthorizeFolder with optional policy

### PR DESCRIPTION
**Pull request by hishamco**
I made this PR to make it easier to authorize the pages or the folder without requiring a policy. It's similar to the `[Authorize]` that we did a lot in many of our applications.
Regarding the code, I was able to make the policy an optional parameter with `null` as default value, but I find the overloading is better in this case

/cc @rynowak

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.RazorPages/DependencyInjection/PageModelConventionBuilderExtensions.cs [hunk 1]
```diff
@@ -23,13 +23,27 @@ public static IPageModelConventionBuilder AllowAnonymousToFolder(this IPageModel
             return builder;
         }
 
+        public static IPageModelConventionBuilder AuthorizePage(this IPageModelConventionBuilder builder, string path)
+        {
+            var filter = new ConstructableAuthorizationFilter(new AuthorizeAttribute());
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/DependencyInjection/PageModelConventionBuilderExtensions.cs]: @rynowak one more reason to update to 1.1 Mvc 😸

**Comment by hishamco** [comment]
@rynowak  should I move this PR to Mvc repo?

**Comment by hishamco** [comment]
Any updates on this?
/cc @rynowak @pranavkm

**Comment by pranavkm** [comment]
@hishamco - the work for RazorPages is happening in https://github.com/aspnet/Mvc. We haven't ported over all the code over so you wouldn't be able to send a PR as yet. That said, feel free to add a work item tracking this.

**Comment by hishamco** [comment]
I will create a new PR there

**Comment by hishamco** [comment]
@pranavkm is `PageModelConventionBuilderExtensions` removed in MVC repo?

**Pull request closed without merging**